content/en/aws/kms/index.md

```diff
@@ -0,0 +1,123 @@
+---
+title: "Key Management Service (KMS)"
+linkTitle: "Key Management Service (KMS)"
+categories: ["LocalStack Community"]
+description: >
+  Key Management Service (KMS)
+---
+
+Key Management Service (KMS) is mainly about
+- creation and management of cryptographic keys
+- use of these keys to encrypt / decrypt data or to sign messages and to verify signatures for signed ones
+
+AWS documentation on KMS and all its operations can be found [here](https://docs.aws.amazon.com/kms/latest/APIReference/Welcome.html).
+
+Keys in KMS roughly fall into the 3 categories:
+- Asymmetric encryption keys - holding a pair of a private and a public key. Asymmetric keys can be used either for data encryption and decryption with [Encrypt](https://docs.aws.amazon.com/kms/latest/APIReference/API_Encrypt.html) / [Decrypt](https://docs.aws.amazon.com/kms/latest/APIReference/API_Decrypt.html) or for creation and verification of digital signatures for messages with [Sign](https://docs.aws.amazon.com/kms/latest/APIReference/API_Sign.html) / [Verify](https://docs.aws.amazon.com/kms/latest/APIReference/API_Verify.html) operations. The same asymmetric key can't be used for both pairs of operations at the same time - when you create such a key, you have to specify in its [KeyUsage](https://docs.aws.amazon.com/kms/latest/APIReference/API_CreateKey.html#KMS-CreateKey-request-KeyUsage) parameter, which pair of operations the key will support.
+- Symmetric encryption keys - such keys consist of just one key. Symmetric keys can **not** be used to sign and to verify messages, are only used to encrypt and to decrypt data.
+- Hash-Based Message Authentication Code (HMAC) keys, a special case of symmetric keys that are only used in creation and verification of message authentication codes (MACs) with [GenerateMac](https://docs.aws.amazon.com/kms/latest/APIReference/API_GenerateMac.html) / [VerifyMac](https://docs.aws.amazon.com/kms/latest/APIReference/API_VerifyMac.html) operations. You can read more about this type of keys [here](https://docs.aws.amazon.com/kms/latest/developerguide/hmac.html).
+
+You can check [this page](https://docs.aws.amazon.com/kms/latest/developerguide/symm-asymm-compare.html) to see which other operations are supported for each type of key. Once you know which operations you require, this reference can help to figure out which key types are suitable for your goals.
+
+# Tutorial
```

Comment:
Please re-name this to `Getting started`. We would have a dedicated tutorial section coming up in the future.

Reply:
I will consider this, if you explain to me what would be the difference between "Getting started" and "Tutorial" sections.

Reply:
A tutorial serves as a step-by-step walkthrough for folks who wish to accomplish a certain thing with LocalStack, usually have no idea of how to do it, and see a tutorial that exactly helps them with the given purpose. Tutorials are also intended to document complex workflows, like most of our Pro samples, which can touch on possible use cases.

Your content falls more in the scope of the "Getting started" guide, where we give enough basic information for someone to do something simplistic, in this case encrypting/decrypting something.

Reply:
This still sounds quite confusing to me. It seems that both tutorials and "getting started" guides are supposed to be step-by-step walkthroughs. The difference between them seems to be not that much in style, rather in scope / difficulty.

I believe that to our customers this is going to be a very confusing approach. Because the names of such sections - "Tutorial" / "Getting started" - do not provide enough information to understand the difference between them. Because the content is going to be very similar in style (step-by-step instructions). And because having multiple such set-by-steps on the same page would result in too big pages, that are going to be confusing and intimidating.

I would propose the following:
- We have just a single step-by-step set of instructions per service page. In some rare cases, where such instructions are like 2-3 commands, we might have 2 such step-by-steps.
- We create a separate section of documentation (separate from docs on services), where we collect and try to structure instructions on any complex scenarios our customers become interested in. We can discuss separately how we organize that.